### ISSUE TITLE
XEP-0313: Message Archive Management

### ISSUE THREAD

**Issue by tomhoefer** [xmppjs/xmpp.js#773]
Hi all,

does this library support Message Archive Management? https://xmpp.org/extensions/xep-0313.html

If yes - is there any documentation available how to use this extension? Unfortunately I haven´t found any documentation

**Comment by tomhoefer** [comment]
One more question: 
Does this library support XEP-0059 (Result Set Management) https://xmpp.org/extensions/xep-0059.html as well?

**Comment by sonnyp** [comment]
Not at the moment.

**Comment by pravi** [comment]
@sonnyp we'd like to crowd fund this feature, would it be possible to help with creating an open collective account? I have found someone who is interested to work on it (I was a mentor for Outreachy with Debian and the interns who worked with me to write a yarn plugin for apt is interested to implement this feature). They are from Nigeria so none of the crowd funding platforms supports paying them directly. We contacted Open Collective and they suggested we ask you.

**Comment by sonnyp** [comment]
What do you need XEP-0313 for?

It's a bit tricky; I'm not sure yet XEP-0313 belongs in `xmpp.js`. Someone (perhaps same initiative?) got in touch https://github.com/xmppjs/xmpp.js/discussions/920

**Comment by pravi** [comment]
@sonnyp thanks for this explanation. It was not clear from earlier comments on this issue. We will try to implement it directly in matrix bifrost.

**Comment by pravi** [comment]
Yes, our goal is to add offline messages to matrix bifrost bridge and we thought adding it in xmpp.is would be best. But as you explained if it is not possible, we will implement it outside xmpp.js.

**Comment by sonnyp** [comment]
Maybe you can ask matrix.org for help with funding / crowdfunding ? 

I have limited availability but happy to review or help with questions. I can also implement or mentor for anything potentially missing in `xmpp.js`.

cc @thesirtaylor

**Comment by pravi** [comment]
We mentioned about the crowd funding plan on https://github.com/matrix-org/matrix-bifrost/issues/64 and also asked someone working with matrix.org We will probably send a direct email as well.
